model.js

```diff
@@ -117,4 +90,42 @@ exports.fetchAllUsers = () => {
   });
 };
 
-// INSERT INTO comments (article_id,author,body) VALUES (1,'lurker','this is a comment') RETURNING *
+exports.updateArticleObj = async () => {
+  articleId = Number(articleId);
+
+  const query = "";
+  return db.query(query).then((data) => {
+    return data.rows;
+  });
+};
+
+exports.validateArticleId = async (articleId) => {
+  articleId = Number(articleId);
+  const allArticle = await this.fetchArticles();
+  const validArticldId = allArticle.map(({ article_id }) => {
+    return article_id;
+  });
+  if (Number.isInteger(articleId) === false) {
+    return Promise.reject({ msg: "invalid input" });
+  } else if (!validArticldId.includes(articleId)) {
+    return Promise.reject({ msg: "Non-existent id" });
+  } else {
+    return articleId;
+  }
+};
+
+exports.updateArticleVote = (articleId, newVote) => {
+  newVote = Number(newVote);
+  const inputData = [[newVote, articleId]];
+  if (!Number.isInteger(newVote)) {
```

Comment:
Same as above, lets allow psql to throw an error if it's going to. Remember that our tables have data type constraints so what happens if our query violates them?